pyodata/v2/model.py

```diff
@@ -439,18 +455,87 @@ def from_literal(self, value):
 
         value = super(EdmDateTimeTypTraits, self).from_literal(value)
 
+        # Note: parse_datetime_literal raises a PyODataModelError exception on invalid formats
+        return parse_datetime_literal(value).replace(tzinfo=current_timezone())
+
+
+class EdmDateTimeOffsetTypTraits(EdmPrefixedTypTraits):
+    """Emd.DateTimeOffset traits
+
+       Represents date and time, plus an offset in minutes from UTC, with values ranging from 12:00:00 midnight,
+       January 1, 1753 A.D. through 11:59:59 P.M, December 9999 A.D
+
+       Literal forms:
+       datetimeoffset'yyyy-mm-ddThh:mm[:ss]±ii:nn' (works for all time zones)
+       datetimeoffset'yyyy-mm-ddThh:mm[:ss]Z' (works only for UTC)
+       NOTE: Spaces are not allowed between datetimeoffset and quoted portion.
+       The datetime part is case-insensitive, the offset one is not.
+
+       Example 1: datetimeoffset'1970-01-01T00:00:01+00:30'
+        - /Date(1000+0030)/ (As DateTime, but with a 30 minutes timezone offset)
+       Example 1: datetimeoffset'1970-01-01T00:00:01-00:60'
+        - /Date(1000-0030)/ (As DateTime, but with a negative 60 minutes timezone offset)
+       https://blogs.sap.com/2017/01/05/date-and-time-in-sap-gateway-foundation/
+    """
+
+    def __init__(self):
+        super(EdmDateTimeOffsetTypTraits, self).__init__('datetimeoffset')
+
+    def to_literal(self, value):
+        """Convert python datetime representation to literal format"""
+
+        if not isinstance(value, datetime.datetime) or value.utcoffset() is None:
+            raise PyODataModelError(
+                f'Cannot convert value of type {type(value)} to literal. Datetime format including offset is required.')
+
+        return super(EdmDateTimeOffsetTypTraits, self).to_literal(value.isoformat())
+
+    def to_json(self, value):
+        if isinstance(value, str):
+            return value
+
+        return f'/Date({int(value.timestamp()) * 1000}{value.utcoffset(value):+})/'
+
+    def from_json(self, value):
+        if value is None:
+            return None
+
+        matches = re.match(r"^/Date\((?P<milliseconds_since_epoch>-?\d+)(?P<offset_in_minutes>[+-]\d+)\)/$", value)
```

Comment:
I am just asking.. not even want to search to prove a point - regexp instead of something from datetime package is really the only option for parsing datetimeoffset? Just that the line itself looks like many tests should be covering this - I know we have re.match usages in the package as well, just that with regexp for entire line I recollect the old joke "now I have X+1 problems" :)

Reply:
Not a big fan either, but the only alternatives I could think of are:
- Something ["scanf()"](https://docs.python.org/3/library/re.html#simulating-scanf) based wrapped in some try or if-else clauses
- Thought of but not working: [strftime](https://docs.python.org/3/library/time.html#time.strftime)

Reply:
Thanks for explanation.